process/nodeStatusProcessor.go

```diff
@@ -246,7 +246,7 @@ func (nsp *NodeStatusProcessor) GetRatingsConfig() (*data.GenericAPIResponse, er
 		return nil, err
 	}
 
-	var responseRatingsConfig data.GenericAPIResponse
+	responseRatingsConfig := &data.GenericAPIResponse{}
```

Comment:
double `&`

I think it would be better to refactor this way:

declare the variable `var response ResponseType` and pass the reference to the endpoint call functions (`&response`)